firefox-ios/firefox-ios-tests/Tests/StorageTests/RustAutofillTests.swift

```diff
@@ -35,229 +35,200 @@ class RustAutofillTests: XCTestCase {
+                    return
                 }
-
-            case .failure(let error):
-                XCTFail("Failed to add address, Error: \(String(describing: error))")
-                expectationAddAddress.fulfill()
-                return
+                continuation.resume(returning: card)
             }
         }
-
-        waitForExpectations(timeout: 3, handler: nil)
     }
 
-    func testListAllAddressesSuccess() {
-        let expectationListAddresses = expectation(description: "Completes the list all addresses operation")
-
-        autofill.listAllAddresses { addresses, error in
-            XCTAssertNil(error, "Error should be nil")
-            XCTAssertNotNil(addresses, "Addresses should not be nil")
-
-            // Assert on individual addresses in the list
-            for address in addresses ?? [] {
-                XCTAssertEqual(address.name, "Jane Doe")
-                XCTAssertEqual(address.streetAddress, "123 Second Avenue")
-                XCTAssertEqual(address.addressLevel2, "Chicago, IL")
-                XCTAssertEqual(address.country, "United States")
+    func updateCreditCard(id: String,
+                          creditCard: UnencryptedCreditCardFields) async throws -> Bool {
+        return try await withCheckedThrowingContinuation { continuation in
+            autofill.updateCreditCard(id: id, creditCard: creditCard) { success, error in
+                guard let success else {
+                    continuation.resume(throwing: error ?? NSError(domain: "Couldn't update credit card", code: 0))
+                    return
+                }
+                continuation.resume(returning: success)
             }
-
-            expectationListAddresses.fulfill()
         }
-
-        waitForExpectations(timeout: 3, handler: nil)
     }
 
-    func testAddCreditCard() {
-        let expectationAddCard = expectation(description: "completed add card")
-        let expectationGetCard = expectation(description: "completed getting card")
+    func listCreditCards() async throws -> [CreditCard] {
+        return try await withCheckedThrowingContinuation { continuation in
+            autofill.listCreditCards { cards, error in
+                guard let cards else {
+                    continuation.resume(throwing: error ?? NSError(domain: "Couldn't list credit cards", code: 0))
+                    return
+                }
+                continuation.resume(returning: cards)
+            }
+        }
+    }
 
-        addCreditCard { creditCard, err in
-            XCTAssertNotNil(creditCard)
-            XCTAssertNil(err)
-            expectationAddCard.fulfill()
+    func deleteCreditCard(id: String) async throws -> Bool {
+        return try await withCheckedThrowingContinuation { continuation in
+            autofill.deleteCreditCard(id: id) { success, error in
+                guard let error else {
```

Comment:
This was the only condition that I needed to check the error first, because the boolean type of `success` isn't an optional.